catalog/app/components/FileEditor/QuiltConfigEditor/BucketPreferences/BucketPreferences.tsx

```diff
@@ -0,0 +1,358 @@
+import type { ErrorObject } from 'ajv'
+import cx from 'classnames'
+import React from 'react'
+import * as M from '@material-ui/core'
+import * as Lab from '@material-ui/lab'
+
+import JsonValidationErrors from 'components/JsonValidationErrors'
+import { docs } from 'constants/urls'
+import * as BucketConfig from 'utils/BucketConfig'
+import StyledLink from 'utils/StyledLink'
+
+import type { QuiltConfigEditorProps } from '../QuiltConfigEditor'
+
+import PackageDescription from './PackageDescription'
+import { parse, stringify } from './State'
+import type { Config, TypedValue, KeyedValue } from './State'
+
+function InputBoolean({
+  className,
+  disabled,
+  onChange,
+  size,
+  value: { key, value },
+}: FieldProps<TypedValue<boolean>>) {
+  const handleChange = React.useCallback(
+    (_e, checked: boolean) => onChange({ isDefault: false, key, value: checked }),
+    [key, onChange],
+  )
+  return (
+    <M.FormControl className={className}>
+      <M.FormControlLabel
+        control={<M.Checkbox checked={!!value} size={size} onChange={handleChange} />}
+        disabled={disabled}
+        label={i18n(key)}
+      />
+    </M.FormControl>
+  )
+}
+
+function InputValue({
+  className,
+  onChange,
+  value: { key, value },
+  ...props
+}: FieldProps<TypedValue<string>>) {
+  const handleChange = React.useCallback(
+    (event) => onChange({ isDefault: false, key, value: event.target.value }),
+    [key, onChange],
+  )
+  return (
+    <M.TextField
+      {...props}
+      className={className}
+      label={i18n(key)}
+      value={value}
+      onChange={handleChange}
+    />
+  )
+}
+
+function InputDefaultSourceBucket({
+  config,
+  onChange,
+  value: { key, value },
+  size,
+  className,
+  disabled,
+}: FieldPropsWithConfig<TypedValue<string>>) {
+  const options = config['ui.source_buckets'].value
+  const handleChange = React.useCallback(
+    (event) => onChange({ isDefault: false, value: event.target.value as string, key }),
+    [key, onChange],
+  )
+  return (
+    <M.FormControl
+      className={className}
+      disabled={!options.length || disabled}
+      fullWidth
+      size={size}
+    >
+      <M.InputLabel>Default source bucket</M.InputLabel>
```

Comment:
```suggestion
      <M.InputLabel>Default bucket</M.InputLabel>
```

"source" is implied by the context